### ISSUE TITLE
Documentation incorrectly states that the room list always returns an empty list if disabled, while federation actually returns a 403 error

### ISSUE THREAD

**Issue by MTRNord** [matrix-org/synapse#13102]
### Description

https://github.com/matrix-org/synapse/blob/develop/docs/usage/configuration/config_documentation.md?plain=1#L3337-L3339 states that both internally and externally synapse will return an empty list on _ALL_ queries.

However https://github.com/matrix-org/synapse/blob/c52abc1cfdd9e5480cdb4a03d626fe61cacc6573/synapse/federation/transport/client.py#L491 (and testing it via fed) states that actually M_FORBIDDEN and a 403 error is returned.

### Steps to reproduce

- Access a remote room list via fed where the remote server runs default config.

### Homeserver

Not relevant. This is a docs issue.

### Synapse Version

1.61.0

### Installation Method

Docker (matrixdotorg/synapse)

### Platform

Not relevant. This is a docs issue.

### Relevant log output

```shell
Not relevant. This is a docs issue.
```

**Comment by MTRNord** [comment]
@deepbluev7 Mentioned also that this might actually be an implementation bug, as https://spec.matrix.org/v1.3/server-server-api/#public-room-directory states that the docs mention the correct thing and 403 is not a valid response from the federation endpoint on fed side. (It's also not a valid response on client side according to https://spec.matrix.org/v1.3/client-server-api/#get_matrixclientv3publicrooms )

**Comment by DMRobertson** [comment]
I think there are a few things going on here; I don't think your diagnosis is completely correct.

The wording in question is 

```
### `enable_room_list_search`

Set to false to disable searching the public room list. When disabled
blocks searching local and remote room lists for local and remote
users by always returning an empty list for all queries. Defaults to true. 
```

Where does this config value get used? I see three relevant reads:

https://github.com/matrix-org/synapse/blob/a164a46038b0e51142781619db0e6dec8e0c2aaa/synapse/handlers/directory.py#L453-L457

https://github.com/matrix-org/synapse/blob/888a29f4127723a8d048ce47cff37ee8a7a6f1b9/synapse/handlers/room_list.py#L86-L87

https://github.com/matrix-org/synapse/blob/888a29f4127723a8d048ce47cff37ee8a7a6f1b9/synapse/handlers/room_list.py#L373-L374

So AFAICS the comment accurately describes the behaviour of calls to `/publicRooms` (the second two snippets).

The first snippet corresponds to `/directory/list/room/<room ID>` which is a different endpoint. (See https://spec.matrix.org/v1.3/client-server-api/#get_matrixclientv3directorylistroomroomid and https://spec.matrix.org/v1.3/client-server-api/#put_matrixclientv3directorylistroomroomid). On the PUT version the spec notes

> Servers may choose to implement additional access control checks here, for instance that room visibility can only be changed by the room creator or a server administrator.

which sounds reasonable. I think one could argue that this wording makes the 403 return code okay. (An aside for @matrix-org/spec-core-team: it is never clear to me if a list of status codes in the spec is normative and/or exhaustive. I think in practice not, because servers can fall over and return 500 at any time.)

In passing, I note that synapse supports a `DELETE  /directory/list/room/<room ID>` endpoint which appears to be unspecced.

https://github.com/matrix-org/synapse/blob/e24ff8ebe3d4119d377355402245947f7de61c00/synapse/rest/client/directory.py#L157-L166

**Comment by DMRobertson** [comment]
TL;DR I don't think Synapse is doing anything obviously wrong (but I'm happy to discuss if you're not convinced).

We should update the docs to better communicate that:

- the empty list trick applies to `/publicRooms` calls
- disabling this option means that Synapse will refuse to modify the **published** room list
- the abstract public room list is a separate concept to the **published** room list

Ugh. The situation is a bit of a mess, isn't it?
